clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -150,35 +200,44 @@ public void testStartupAndTearDown() throws InterruptedException {
     }
 
     @Test
-    public void testApplicationEvent() {
-        ApplicationEvent e = new PollEvent(100);
-        applicationEventsQueue.add(e);
+    public void testRequestManagersArePolledOnce() {
+        List<Optional<? extends RequestManager>> list = new ArrayList<>();
+        list.add(Optional.of(coordinatorRequestManager));
+        list.add(Optional.of(heartbeatRequestManager));
+        list.add(Optional.of(offsetsRequestManager));
+
+        when(requestManagers.entries()).thenReturn(list);
+        when(coordinatorRequestManager.poll(anyLong())).thenReturn(mock(NetworkClientDelegate.PollResult.class));
         consumerNetworkThread.runOnce();
-        verify(applicationEventProcessor, times(1)).process(e);
+        requestManagers.entries().forEach(rmo -> rmo.ifPresent(rm -> verify(rm).poll(anyLong())));
+        requestManagers.entries().forEach(rmo -> rmo.ifPresent(rm -> verify(rm).maximumTimeToWait(anyLong())));
+        verify(networkClientDelegate).addAll(any(NetworkClientDelegate.PollResult.class));
+        verify(networkClientDelegate).poll(anyLong(), anyLong());
     }
 
-    @Test
-    public void testMetadataUpdateEvent() {
-        ApplicationEvent e = new NewTopicsMetadataUpdateRequestEvent();
-        applicationEventsQueue.add(e);
-        consumerNetworkThread.runOnce();
-        verify(metadata).requestUpdateForNewTopics();
-    }
-
-    @Test
-    public void testAsyncCommitEvent() {
-        ApplicationEvent e = new AsyncCommitEvent(new HashMap<>());
+    @ParameterizedTest
+    @MethodSource("appEvents")
+    public void testEventIsProcessed(ApplicationEvent e) {
         applicationEventsQueue.add(e);
         consumerNetworkThread.runOnce();
-        verify(applicationEventProcessor).process(any(AsyncCommitEvent.class));
+        verify(applicationEventProcessor).process(any(e.getClass()));
```

Comment:
this is not right, the method call is `applicationEventProcessor.process();` in runOnce()